src/lava/networks/gradedvecnetwork.py

```diff
@@ -0,0 +1,273 @@
+
+    def __init__(self, vec, **kwargs):
+
+        self.loihi2 = kwargs.pop('loihi2', False)
+        self.shape = np.atleast_2d(vec).shape
+        self.exp = kwargs.pop('exp', 0)
+        print(self.shape)
+
+        # convert it to fixed point base
+        vec *= 2**self.exp
+
+        self.inport_plug = source.RingBuffer(data=np.atleast_2d(vec))
+
+        if self.loihi2:
+            self.inport_adapter = eio.spike.PyToNxAdapter(
+                shape=(self.shape[0],),
+                num_message_bits=24)
+            self.inport_plug.s_out.connect(self.inport_adapter.inp)
+            self.out_port = self.inport_adapter.out
+
+        else:
+            self.out_port = self.inport_plug.s_out
+
+    def __lshift__(self, other):
+        # maybe this could be done with a numpy array and call set_data?
+        return NotImplemented
+
+
+class OutputVec(AlgebraicVector):
+    """OutputVec
+    Records spike output. Adds algebraic syntax to RingBuffer
+
+    Parameters
+    ----------
+    shape=(1,): tuple(int)
+        shape of the output to record
+    buffer=1: int
+        length of the recording.
+        (buffer is overwritten if shorter than sim time).
+    loihi2=False: bool
+        Flag to create the adapters for loihi 2.
+    num_message_bits=24: int
+        size of output message. ("0" is for unary spike event).
+    """
+
+    def __init__(self, **kwargs):
+
+        self.shape = kwargs.pop('shape', (1,))
+        self.buffer = kwargs.pop('buffer', 1)
+        self.loihi2 = kwargs.pop('loihi2', False)
+        self.num_message_bits = kwargs.pop('num_message_bits', 24)
+
+        self.outport_plug = sink.RingBuffer(
+            shape=self.shape, buffer=self.buffer, **kwargs)
+
+        if self.loihi2:
+            self.outport_adapter = eio.spike.NxToPyAdapter(
+                shape=self.shape, num_message_bits=self.num_message_bits)
+            self.outport_adapter.out.connect(self.outport_plug.a_in)
+            self.in_port = self.outport_adapter.inp
+        else:
+            self.in_port = self.outport_plug.a_in
+
+    def get_data(self):
+        return (self.outport_plug.data.get().astype(np.int32) << 8) >> 8
+
+
+class LIFVec(AlgebraicVector):
+    """LIFVec
+    Network wrapper to LIF neuron.
+
+    Parameters
+    ----------
+    See lava.proc.lif.process.LIF
+    """
+
+    def __init__(self, **kwargs):
+        self.main = LIF(**kwargs)
+
+        self.in_port = self.main.a_in
+        self.out_port = self.main.s_out
+
+
+class ThreshVec(AlgebraicVector):
+    """ThreshVec
+    Simple graded threshold vector with no dynamics.
+
+    Parameters
+    ----------
+    shape=(1,): tuple(int)
+        Number and topology of neurons.
+    vth=10: int
+        Threshold for spiking.
+    exp=0: int
+        Fixed point base of the vector.
+    """
+
+    def __init__(self, **kwargs):
+
+        self.shape = kwargs.pop('shape', (1,))
+        self.vth = kwargs.pop('vth', 10)
+        self.exp = kwargs.pop('exp', 0)
```

Comment:
See comments about docstrings, `__init__` signature and type hints I added for `InputVec`.